modules/development/src/main/java/unnamed/mmo/server/dev/blocks/ore/OreCreatorTool.java

```diff
@@ -0,0 +1,109 @@
+import net.minestom.server.entity.Player;
+import net.minestom.server.instance.Instance;
+import net.minestom.server.inventory.Inventory;
+import net.minestom.server.inventory.InventoryType;
+import net.minestom.server.inventory.click.ClickType;
+import net.minestom.server.inventory.condition.InventoryConditionResult;
+import net.minestom.server.item.ItemStack;
+import net.minestom.server.item.Material;
+import net.minestom.server.tag.Tag;
+import net.minestom.server.utils.NamespaceID;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import unnamed.mmo.blocks.ore.Ore;
+import unnamed.mmo.server.dev.tool.DebugTool;
+import unnamed.mmo.util.ComponentUtil;
+
+import java.util.Comparator;
+import java.util.List;
+
+@AutoService(DebugTool.class)
+public class OreCreatorTool implements DebugTool {
+    private static final Tag<String> SELECTION_TAG = Tag.String("ore_creator/selection");
+
+    @Override
+    public @NotNull NamespaceID namespace() {
+        return NamespaceID.from("unnamed:ore_creator");
+    }
+
+    @Override
+    public @NotNull ItemStack itemStack() {
+        return ItemStack.builder(Material.DIAMOND)
+                .displayName(ComponentUtil.fromStringSafe("Ore Creator"))
+                .lore(
+                        ComponentUtil.fromStringSafe("<white>Selection: None"),
+                        ComponentUtil.fromStringSafe("<white>Shift+RMB: Select ore"),
+                        ComponentUtil.fromStringSafe("<white>RMB: Assign selected ore")
+                )
+                .build();
+    }
+
+    @Override
+    public @NotNull ItemStack rightClicked(@NotNull Player player, @NotNull ItemStack itemStack, @Nullable Point targetBlock, @Nullable Entity targetEntity) {
+        return player.isSneaking() ?
+                openOreSelector(player, itemStack) :
+                assignOre(player, itemStack, targetBlock);
+    }
+
+    private ItemStack assignOre(Player player, ItemStack itemStack, Point targetBlock) {
+        if (targetBlock == null)
+            return itemStack;
+
+        final Ore ore = Ore.fromNamespaceId(itemStack.getTag(SELECTION_TAG));
+        if (ore == null) {
+            player.sendMessage("You must select an ore before placing it.");
+            return itemStack;
+        }
+
+        final Instance instance = player.getInstance();
+        instance.setBlock(targetBlock, ore.asBlock());
+        player.sendMessage("Set " + targetBlock + " to " + ore.name() + "!");
+        return itemStack;
+    }
+
+    private ItemStack openOreSelector(Player player, ItemStack itemStack) {
+        final Inventory oreSelector = new Inventory(InventoryType.CHEST_2_ROW, "Ore Selector");
+        oreSelector.addInventoryCondition(this::selectOre);
+        Ore.REGISTRY.values().stream()
+                .sorted(Comparator.comparing(Ore::name))
+                .forEach(ore -> {
+                    //todo is there a better Block->Material conversion
```

Comment:
`ore.oreBlock().registry().material()`? Though I don't know how well that will work.